src/bin/pg_autoctl/keeper.c

```diff
@@ -113,6 +113,53 @@ keeper_update_state(Keeper *keeper, int node_id, int group_id,
 }
 
 
+/*
+ * keeper_should_ensure_current_state returns true when pg_autoctl should
+ * ensure that Postgres is running, or not running, depending on the current
+ * FSM state, before calling the transition function to the next state.
+ *
+ * At the moment, the only cases when we DON'T want to ensure the current state
+ * are when either the current state or the goal state are one of the following:
+ *
+ *  - DRAINING
+ *  - DEMOTED
+ *  - DEMOTE TIMEOUT
+ *
+ * That's because we would then stop Postgres first when going from DEMOTED to
+ * SINGLE, or ensure Postgres is running when going from PRIMARY to DEMOTED.
+ * This last example is a split-brain hazard, too.
+ */
+bool
+keeper_should_ensure_current_state_before_transition(Keeper *keeper)
+{
+	KeeperStateData *keeperState = &(keeper->state);
+
+	if (keeperState->assigned_role != keeperState->current_role)
```

Comment:
can we something like below to make it more explicit. 
```c
if (keeperState->assigned_role == keeperState->current_role)
  return false
```

Reply:
Fair enough, done now.